torch/_inductor/codegen/common.py

```diff
@@ -53,16 +54,111 @@ def data_type_logger(msg):
         schedule_log.debug("Data type propagation: %s", msg)
 
 
-@dataclasses.dataclass
+class WorkspaceZeroMode(enum.Enum):
+    UNINITIALIZED = 0
+    ZERO_ON_CALL = 1  # kernel may leave workspace dirty
+    ZERO_PER_GRAPH = 2  # must be re-zeroed by kernel
+
+    @staticmethod
+    def combine(a, b):
+        if a == b or b == WorkspaceZeroMode.UNINITIALIZED:
+            return a
+        if a == WorkspaceZeroMode.UNINITIALIZED:
+            return b
+        raise NotImplementedError(f"WorkspaceZeroMode.combine({a!r}, {b!r})")
+
+    @staticmethod
+    def from_bool(zero_fill):
+        if zero_fill:
+            return WorkspaceZeroMode.ZERO_ON_CALL
+        return WorkspaceZeroMode.UNINITIALIZED
+
+
+@dataclasses.dataclass(frozen=True)
```

Comment:
I did just (5 minutes ago) land https://github.com/pytorch/pytorch/pull/137859/files#diff-cf6ca00beddc32a2a6a2933fb9913b6a2b925ffc3b745488967210e4343134acR113, so you can use `ir_dataclass`

Reply:
Would be circular import :(

Reply:
Should we put this code in `ir.py`? It's basically an IRNode, no?

Reply:
Sort of... I just moved ir_dataclass to utils.py.